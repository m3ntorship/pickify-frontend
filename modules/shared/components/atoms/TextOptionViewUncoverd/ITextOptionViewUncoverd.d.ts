declare namespace ITextOptionViewUncoverd {
  export interface IProps extends IData {
    letter?: string;
    percentage?: number;
    mostVoted?: boolean;
    isExpanded: boolean;
    type: string;
  }
  export interface IData {
    id: string;
    optionBody: string;
    isOptionChecked?: boolean;
  }
}

export { ITextOptionViewUncoverd };
