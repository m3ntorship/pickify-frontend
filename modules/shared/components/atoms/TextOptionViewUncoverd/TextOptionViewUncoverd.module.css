.btnBody {
  @apply flex justify-between items-center rounded-md p-xs pr-xsvv text-dark;
}
.letter {
  @apply bg-white text-dark-grey font-normal capitalize rounded-sm mr-3 w-l h-l flex justify-center items-center;
}
.option {
  @apply text-sm capitalize whitespace-nowrap;
}
.percentage {
  @apply font-normal text-sm flex items-center z-10;
}
.flex-container {
  @apply flex items-center mr-4 z-10;
}
.mostVoted {
  @apply bg-primary-shd3;
}
.leastVoted {
  @apply bg-primary-shd7;
}
.goldenStarIcon {
  @apply w-4 h-5 mr-1;
}
