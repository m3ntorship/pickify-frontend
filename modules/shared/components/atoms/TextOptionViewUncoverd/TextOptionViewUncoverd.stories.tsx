import type { ReactElement } from 'react';
import type { Story } from '@storybook/react';
import TextOptionViewUncoverd from './TextOptionViewUncoverd';
import type { ITextOptionViewUncoverd } from './ITextOptionViewUncoverd';

export default {
  title: 'Atoms/TextOptionViewUncoverd',
  component: TextOptionViewUncoverd,
  argTypes: {},
};

const Template: Story<ITextOptionViewUncoverd.IProps> = (
  args,
): ReactElement => <TextOptionViewUncoverd {...args} />;
export const Default = Template.bind({});
Default.args = {
  optionBody: 'option 1',
  letter: 'A',
  percentage: 25,
  isOptionChecked: false,
  mostVoted: true,
};
