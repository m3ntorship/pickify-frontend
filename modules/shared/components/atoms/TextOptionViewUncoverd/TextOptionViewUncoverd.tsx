import React from 'react';
import classNames from 'classnames';
import type { FC, ReactElement } from 'react';
import { EPostType } from '@modules/shared/types/postFeed/EPostType';
import type { ITextOptionViewUncoverd } from './ITextOptionViewUncoverd';
import styles from './TextOptionViewUncoverd.module.css';
import Check from '../../icons/checkMarkDefault.svg';
import GoldenStarIcon from '../../icons/goldenStar.svg';

const TextOptionViewUncoverd: FC<ITextOptionViewUncoverd.IProps> = ({
  letter,
  optionBody,
  percentage,
  isOptionChecked,
  mostVoted,
  id,
  isExpanded,
  type,
}): ReactElement => {
  const textPoll = classNames(styles.btnBody, {
    [styles.mostVoted]: mostVoted,
    [styles.leastVoted]: !mostVoted,
  });
  const svgClasses = classNames({
    [styles.svgDark]: !mostVoted,
    [styles.svgWhite]: mostVoted,
  });
  return type !== EPostType.MiniSurvey || isExpanded ? (
    <button
      type="button"
      id={id}
      data-testid={id}
      className={textPoll}
      disabled
    >
      <span className={styles['flex-container']}>
        {letter && <span className={styles.letter}>{letter}</span>}
        <p className={styles.option} dir="auto">
          {optionBody}
        </p>
        <span />
        {isOptionChecked && (
          <span className={styles['svg-check']}>
            <Check className={svgClasses} />
          </span>
        )}
      </span>
      <span className={styles.percentage}>
        {mostVoted ? <GoldenStarIcon className={styles.goldenStarIcon} /> : ''}
        {percentage}%
      </span>
    </button>
  ) : (
    <>
      mostVoted &&
      <p>{optionBody}</p>
    </>
  );
};
export default TextOptionViewUncoverd;
