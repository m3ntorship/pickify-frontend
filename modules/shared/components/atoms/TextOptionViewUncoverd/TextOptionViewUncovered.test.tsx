import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import TextOptionViewUncoverd from './TextOptionViewUncoverd';

describe('Snapshot testing for OptionViewUncoverd component', () => {
  const twenty = 20;
  it('should have class of bg-primary-shd1 while Testing OptionViewUncoverd with mostVoted to true', () => {
    const tree = renderer
      .create(
        <TextOptionViewUncoverd
          id="test1"
          optionBody="option 1"
          letter="A"
          percentage={twenty}
          mostVoted
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have checkmark svg while Testing OptionViewUncoverd with the necessary props with isOptionChecked={true}', () => {
    const tree = renderer
      .create(
        <TextOptionViewUncoverd
          id="test1"
          optionBody="option 1"
          letter="A"
          percentage={twenty}
          isOptionChecked
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Behavioural tests for OptionViewUncoverd component ', () => {
  const twenty = 20;
  it('should render "option 1 " as option body', () => {
    render(
      <TextOptionViewUncoverd id="test1" optionBody="option 1" letter="A" />,
    );
    const testBtn = screen.getByText('option 1');
    expect(testBtn).toBeInTheDocument();
  });

  it('should render "20%" as percentage', () => {
    render(
      <TextOptionViewUncoverd
        id="test1"
        optionBody="option 1"
        percentage={twenty}
        letter="A"
      />,
    );
    const testBtn = screen.getByText('20%');
    expect(testBtn).toBeInTheDocument();
  });

  it('should be disabled  while its uncovered', () => {
    render(
      <TextOptionViewUncoverd id="test1" optionBody="option 1" letter="A" />,
    );

    const testBtn = screen.getByTestId('test1');
    expect(testBtn).toBeDisabled();
  });
});
