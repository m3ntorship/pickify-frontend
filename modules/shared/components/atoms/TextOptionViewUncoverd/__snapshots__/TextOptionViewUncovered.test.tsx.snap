// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing for OptionViewUncoverd component should have checkmark svg while Testing OptionViewUncoverd with the necessary props with isOptionChecked={true} 1`] = `
<button
  className="btnBody leastVoted"
  data-testid="test1"
  disabled={true}
  id="test1"
  type="button"
>
  <span
    className="flex-container"
  >
    <span
      className="letter"
    >
      A
    </span>
    <p
      className="option"
    >
       
      option 1
       
    </p>
    <span />
    <span
      className="svg-check"
    >
      <svg
        className="svgDark"
        data-file-name="SvgCheckMarkDefault"
      />
    </span>
  </span>
  <span
    className="percentage"
  >
    
    20
    %
  </span>
</button>
`;

exports[`Snapshot testing for OptionViewUncoverd component should have class of bg-primary-shd1 while Testing OptionViewUncoverd with mostVoted to true 1`] = `
<button
  className="btnBody mostVoted"
  data-testid="test1"
  disabled={true}
  id="test1"
  type="button"
>
  <span
    className="flex-container"
  >
    <span
      className="letter"
    >
      A
    </span>
    <p
      className="option"
    >
       
      option 1
       
    </p>
    <span />
  </span>
  <span
    className="percentage"
  >
    <svg
      className="goldenStarIcon"
      data-file-name="SvgGoldenStar"
    />
    20
    %
  </span>
</button>
`;
